src/libraries/System.Runtime/tests/System/Text/StringBuilderTests.cs

```diff
@@ -1727,8 +1727,8 @@ public static void Replace_String_Invalid()
             AssertExtensions.Throws<ArgumentException>("oldValue", () => builder.Replace("", "a")); // Old value is empty
             AssertExtensions.Throws<ArgumentException>("oldValue", () => builder.Replace("", "a", 0, 0)); // Old value is empty
 
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Replace("o", "oo")); // New length > builder.MaxCapacity
-            AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Replace("o", "oo", 0, 5)); // New length > builder.MaxCapacity
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => builder.Replace("o", "oo")); // New length > builder.MaxCapacity
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => builder.Replace("o", "oo", 0, 5)); // New length > builder.MaxCapacity
```

Comment:
The fact that neither "requiredLength" nor "count" are things at this call site really shows that exceeding MaxCapacity isn't an argument exception, but an InvalidOperationException.

I'm almost certain I'm not suggesting to change that right now, though, merely commenting.  (Possibly there's another analyzer here, which is "don't throw ArgumentException from private methods, unless they're helpers"... but I can't describe how to detect a helper.  Though it's quite certainly in the "Design" category.)